### ISSUE TITLE
"InternalServerError: is at commit <N> but expected commit <N - 1>" on image upload

### ISSUE THREAD

**Issue by ifvictr** [ifvictr/ghost-storage-github#21]
Getting this error when trying to upload a cover image to a post.

```
InternalServerError: is at 9222de0514c40164f1fc209ba563eb326b1c6826 but expected 18990c738b7fddfacc3e1c0042d242a97837f3ca
```

**Comment by ifvictr** [comment]
I think the problem might be caused my Ghost trying to upload two images at the same time (original + optimized), leading to some sort of mismatch in the Git hash.

**Comment by ifvictr** [comment]
This issue may be related to #18.

**Comment by ifvictr** [comment]
Possible solution: try to fetch the latest commit before attempting to create a new file.

**Comment by ifvictr** [comment]
Closing because #18 already exists.

**Comment by ifvictr** [comment]
Closed by #34.
